### ISSUE TITLE
Repository size > 50 Mb

### ISSUE THREAD

**Issue by nabobalis** [LM-SAL/aiapy#42]
In GitLab by @wtbarnes on Feb 26, 2020, 09:48

The size of this repository is unacceptably large at >50 Mb. This is because at an early point in the repo history, a set of 4K-by-4K FITS files were added as part of the PSF calculation. Though these have now been removed, they are still part of the history of the repository.

We need to work out how to actually remove these from the history of the repo. See [this link](https://itextpdf.com/en/blog/technical-notes/how-completely-remove-file-git-repository) for a few possible solutions.

**Comment by nabobalis** [comment]
In GitLab by @wtbarnes on Mar 11, 2020, 14:37

@markcheung Could you move your psf-toolkit branch off the main repo and onto your fork?

**Comment by nabobalis** [comment]
In GitLab by @wtbarnes on Mar 11, 2020, 15:11

For reference, this is the offending commit: f39934b9176970e557c10e5a04b9fc3a518e3a66

**Comment by nabobalis** [comment]
In GitLab by @wtbarnes on Mar 11, 2020, 16:01

Also see these [very helpful instructions](https://github.com/sunpy/sunpy/pull/323#issuecomment-9407117) from a similar snafu on the sunpy repo a few years back

After some experimenting with a clean copy of the repo, the following seems to work,

```shell
$ git filter-branch -f --index-filter 'git rm --cached --ignore-unmatch aiapy/psf/data/*.npz' --prune-empty -- f39934b^.. --all
$ git filter-branch -f --index-filter 'git rm --cached --ignore-unmatch response/PSF/*.npz' --prune-empty -- f39934b^.. --all
$ rm -rf .git/refs/original
$ git reflog expire --expire=now --all
$ git gc --prune=now
```

The first command is run twice because I renamed those files at some point

I will need to do this and then force push to the upstream repo I'm afraid. This has the unfortunate side-effect of rewriting the history of every commit since the one that committed those files which is unfortunately the entire history. This also means that everyone who has forked this repo will either need to also run this series of commands or delete their fork and refork the project.

**Comment by nabobalis** [comment]
In GitLab by @wtbarnes on Mar 11, 2020, 16:02

This is really a blocking issue for a release and the JOSS paper. While we could go on without fixing this, the problem will only get more difficult to fix and it is annoying to have to carry around ~50 MB with this repo for no reason.

**Comment by nabobalis** [comment]
In GitLab by @wtbarnes on Mar 11, 2020, 16:15

Before doing this, we should also merge all open PRs.

**Comment by nabobalis** [comment]
In GitLab by @markcheung on Mar 11, 2020, 17:28

I'm not comfortable enough with git to give a preference about which is the best solution.

**Comment by nabobalis** [comment]
In GitLab by @wtbarnes on Mar 11, 2020, 17:38

I think the best solution is running the snippet I included above and force pushing to master. I'm comfortable with doing this and have consulted with a few of the sunpy devs who were in a similar situation. I ran it on a test copy the repository and it reduces the size of the repo to ~0.6 MB.

Before that though, I'd like to remove the `psf-toolkit` branch (it still has the `.npz` files) and merge !48.
